cpp/src/arrow/ipc/writer.cc

```diff
@@ -526,6 +526,10 @@ class RecordBatchSerializer {
     return VisitType(*array.indices());
   }
 
+  Status Visit(const RunEndEncodedArray& type) {
+    return Status::NotImplemented("run-end encoded array in IPC");
+  }
```

Comment:
I hope there will be a followup PR for this?

Reply:
Yes. It needs to be done so compilation can succeed and progress can be made. That makes turning this into small PRs a bit tricky as well.